### DISCUSSION TITLE
How to get route with params from middleware?

### DISCUSSION THREAD

**Discussion by naydichev** [koajs/koa#1801] [Q&A] [answered]
For example, let's say I have the following routes:

- `/users/:id`
- `/users/:id/friends`
- `/users/:id/friends/:friendId`

I see that I can get the `originalUrl` from the context, which will be something like `/users/5` or `/users/5/friends/132`, but what I'd like to get is `/users/:id` or `/users/:id/friends/:friendId`.

Is this possible?

**Comment by fengmk2** [comment, marked as answer]
if you are use koa router (https://github.com/koajs/router/blob/master/lib/router.js#L392), you can access matched and routerPath from ctx.

**Reply by naydichev** [reply]
Thank you - I was having a hard time finding that in the documentation.

**Reply by cfYYDS** [reply]
ctx.request.params.xxx
